src/Plugin/EntityBrowser/SelectionDisplay/NewDisplay.php

```diff
@@ -0,0 +1,162 @@
+        '#default_value' => $ids,
+        // #ajax is officially not supported for hidden elements but if we
+        // specify event manually it works.
+        '#ajax' => [
+          'callback' => [get_class($this), 'updateWidgetCallback'],
+          'wrapper' => $details_id,
+          'event' => 'entity_browser_selected_updated',
+        ],
+      ],
+    ];
+
+    $form['current'] = [
+      '#theme_wrappers' => ['container'],
+      '#attributes' => ['class' => ['selected-entities-list']],
+      'items' => array_map(
+        function($id) use ($selected_entities, $details_id) {
+          $entity = $selected_entities[$id];
+          return [
+            '#theme_wrappers' => ['container'],
+            '#attributes' => [
+              'class' => ['selected-item-container'],
+              'data-entity-id' => $entity->id()
+            ],
+            'display' => ['#markup' => $entity->label()],
+            'remove_button' => [
+              '#type' => 'submit',
+              '#value' => $this->t('Remove'),
+              '#ajax' => [
+                'callback' => [get_class($this), 'updateWidgetCallback'],
+                'wrapper' => $details_id,
+              ],
+              '#submit' => [[get_class($this), 'removeItemSubmit']],
+              '#name' => 'remove_' . $id,
+              '#attributes' => ['data-entity-id' => $id]
+            ],
+          ];
+        },
+        $ids
+      ),
+    ];
+    $form['use_selected'] = array(
+      '#type' => 'submit',
+      '#value' => t('Use selected'),
+      '#name' => 'use_selected',
+    );
+
+    return $form;
+  }
+
+  /**
+   * Submit callback for remove buttons.
+   */
+  public static function removeItemSubmit(&$form, FormStateInterface $form_state) {
+    $triggering_element = $form_state->getTriggeringElement();
+    if (!empty($triggering_element['#attributes']['data-entity-id'])) {
+      $id = $triggering_element['#attributes']['data-entity-id'];
+      $parents = array_slice($triggering_element['#parents'], 0, -4);
+      $array_parents = array_slice($triggering_element['#array_parents'], 0, -4);
+
+      // Find and remove correct entity.
+      $values = explode(' ', $form_state->getValue(array_merge($parents, ['target_id'])));
+      $values = array_filter(
+        $values,
+        function($item) use ($id) { return $item != $id; }
+      );
+      $values = implode(' ', $values);
+
+      // Set new value for this widget.
+      $target_id_element = &NestedArray::getValue($form, array_merge($array_parents, ['target_id']));
+      $form_state->setValueForElement($target_id_element, $values);
+      NestedArray::setValue($form_state->getUserInput(), $target_id_element['#parents'], $values);
+
+      // Rebuild form.
+      $form_state->setRebuild();
+    }
+  }
+
+  /**
+   * AJAX form callback.
+   */
+  public static function updateWidgetCallback(array &$form, FormStateInterface $form_state) {
+    $trigger = $form_state->getTriggeringElement();
```

Comment:
This function was copied from entity reference widget and has a lot of code that makes no sense here. Needs cleanup.